decidim-elections/app/commands/decidim/votings/create_polling_station_results.rb

```diff
@@ -35,10 +40,29 @@ def call
       def create_answer_result_for(answer_result)
         params = {
           decidim_votings_polling_station_id: form.polling_station_id,
+          decidim_elections_election_id: form.election_id,
           votes_count: answer_result.votes_count,
-          decidim_elections_answer_id: answer_result.id
+          decidim_elections_question_id: answer_result.question_id,
+          decidim_elections_answer_id: answer_result.id,
+          result_type: "valid_answer"
+        }
+
+        create_result!(params)
+      end
+
+      def create_question_result_for(question_result)
```

Comment:
💭 Since you use `create!` in `create_result!`, I'd use it for the method as well: `create_question_result_for!`. Same for the `create_answer_result_for!`. 
Also, what happens, if one of these methods fails? Does it make sense to wrap it inside a transaction?

Reply:
Great suggestions, yes to both 😃. 
I'll add a bang `!` to method names, and wrap them in a `transaction`.